### PR TITLE
Refs #35058 -- Added __repr__() to OGRGeomType.

### PR DESCRIPTION
**Pull request by smithdc1**

### CODE REVIEW

**Review by claudep — APPROVED** [review]
:+1:

File: tests/gis_tests/gdal_tests/test_geom.py [hunk 1]
```diff
@@ -58,6 +58,9 @@ def test_geomtype(self):
         self.assertEqual(0, gt.num)
         self.assertEqual("Unknown", gt.name)
 
+    def test_geom_type_repr(self):
+        self.assertEqual(OGRGeomType("point").__repr__(), "<OGRGeomType: Point>")
```
Comment (felixxm) [inline tests/gis_tests/gdal_tests/test_geom.py:62]:
```suggestion
        self.assertEqual(repr(OGRGeomType("point")), "<OGRGeomType: Point>")
```

File: django/contrib/gis/gdal/geomtype.py [hunk 2]
```diff
@@ -55,6 +55,9 @@ def __str__(self):
         "Return the value of the name property."
         return self.name
 
+    def __repr__(self):
+        return f"<{self.__class__.__name__}: {self.name}>"
```
Comment (felixxm) [inline django/contrib/gis/gdal/geomtype.py:59]:
```suggestion
        return f"<{self.__class__.__qualname__}: {self.name}>"
```

**Comment by felixxm** [comment]
@smithdc1 Thanks :+1:

**Pull request merged**